hypershift-operator/main.go

```diff
@@ -244,16 +243,43 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
 		}
 	}
 
+	// The mgr and therefore the cache is not started yet, thus we have to construct a client that
+	// directly reads from the api.
+	apiReadingClient, err := crclient.NewDelegatingClient(crclient.NewDelegatingClientInput{
+		CacheReader: mgr.GetAPIReader(),
+		Client:      mgr.GetClient(),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to construct api reading client: %w", err)
+	}
+
+	// Populate registry overrides with any ICSP and IDMS from a OpenShift management cluster
+	var imageRegistryOverrides map[string][]string
+	if mgmtClusterCaps.Has(capabilities.CapabilityICSP) || mgmtClusterCaps.Has(capabilities.CapabilityIDMS) {
+		// Warn the user this CR will be deprecated in the future
+		if mgmtClusterCaps.Has(capabilities.CapabilityICSP) {
```

Comment:
This will be true for a while (that the ICSP CRD exists). I think you need to log the warning only if you actually detect ICSP cr's